pages/register.tsx

```diff
@@ -0,0 +1,290 @@
+                                    required />
+                            </div>
+                            <div className={styles.lName}>
+                                <TextField
+                                    name="lastName"
+                                    label="Last Name"
+                                    variant="filled"
+                                    onChange={handleChange}
+                                    value={formValues.lastName}
+                                    fullWidth
+                                    required />
+                            </div>
+                        </div>
+
+                        <div className={styles.userDetails}>
+                            <FormControl variant="filled" size="small" fullWidth>
+                                <InputLabel>Branch</InputLabel>
+                                <Select
+                                    name="branch"
+                                    onChange={handleChange}
+                                    value={formValues.branch}
+                                    sx={{ height: "4em" }}
+                                >
+                                    {
+                                        branchNames.map((branchName, id) => <MenuItem value={branchName} key={id}>{branchName}</MenuItem>)
+                                    }
+                                </Select>
+                            </FormControl>
+
+                            <div>
+                                <TextField label="SID"
+                                    name="sid"
+                                    variant="filled"
+                                    error={!isSidCorrect(formValues.sid)}
+                                    onChange={handleChange}
+                                    value={formValues.sid}
+                                    fullWidth
+                                    required />
+                                {
+                                    isSidCorrect(formValues.sid)
+                                        ? <></>
+                                        : <ErrorTextBox text={errorText.invalidSid} icon={<BiErrorCircle />} />
+                                }
+                            </div>
+                        </div>
+                    </div>
+
+                </div>
+
+                {/* User Account Details */}
+                <div className={styles.sectionWrapper}>
+
+                    <div className={styles.flowSection}>
+                        <div className={styles.accountDetails}>
+                            <TextField
+                                name="username"
+                                label="Username"
+                                variant="filled"
+                                onChange={handleChange}
+                                value={formValues.username}
+                                fullWidth
+                                required />
```

Comment:
We are doing just email/password login, username isnt required, but we can keep it if it looks cool like discord

> It wont be used for login.

Reply:
@ken1000minus7 what do you say? Keep or leave?

Reply:
Not required I would say, but if u want can add